binderhub/app.py

```diff
@@ -251,6 +251,51 @@ def _add_slash(self, proposal):
         """
     )
 
+    use_as_federation_portal = Bool(False, config=True,
+        help="""configure the current binder to also act as a federation portal.
+
+        This settings should be used only on mybinder.org deployment or if you
+        have multiple internal private deployment of binderhubs inside a closed
+        network. It exposes a couple of extra endpoints to binderhub for remote
+        services to register themselves. Once register for a specific user, the
+        binder portal will redirect users to these endpoints instead of running
+        them locally.
+        """
+    )
+
+    default_binders_list = List([], config=True,
+        help="""
+        EXPERIMENTAL: A list of known default binder which can be chosen by users.
+
+        Example:
+            c.BinderHub.default_binders_list = [
+                    'https://staging.mybinder.org',
+                    'https://mybinder.org'
+            ]
+        """
+    )
+
+    list_cookie_set_binders = Bool(False, config=True,
+        help="""
+        EXPERIMENTAL: Federation between binder can optionally be done by listing
+        binders in a cookie, which allow users to enable their own binders.
+
+        In some cases this can be undesirable, so disabled by default.
+        """
+    )
+
+
+    cannonical_address = Unicode('', config=True,
```

Comment:
`federation_site_address`

```
help="""Canonical address of a federation site. This address is used by
the federation portal.

Example:

Trailing slash is required.
"""

Wasn't sure what this statement means "When current instance is used as a portal also used to avoid redirect loops to self."